pkg/serverless/trace/propagation/carriers_test.go

```diff
@@ -169,22 +169,22 @@ func TestSnsSqsMessageCarrier(t *testing.T) {
 				}`,
 			},
 			expMap: nil,
-			expErr: errors.New("Error unmarshaling message body: json: cannot unmarshal string into Go struct field .MessageAttributes of type map[string]struct { Type string; Value string }"),
+			expErr: errors.New("Error unmarshaling message body: json: cannot unmarshal string into Go struct field SNSEntity.MessageAttributes of type map[string]interface {}"),
 		},
 		{
 			name: "non-binary-type",
 			event: events.SQSMessage{
 				Body: `{
 					"MessageAttributes": {
 						"_datadog": {
-							"Type": "String",
+							"Type": "Purple",
```

Comment:
I followed sns trace extraction from python where we also support extracting from String types in addition to Binary types.